File: leetcode/editor/cn/387_字符串中的第一个唯一字符_test.go
package leetcode

import (
	"github.com/stretchr/testify/assert"
	"testing"
)

func Test_givenNormal_when387_thenSuccess(t *testing.T) {
	assert.Equal(t, 2, firstUniqChar("loveleetcode"))
}

//leetcode submit region begin(Prohibit modification and deletion)
func firstUniqChar(s string) int {
	ans := -1
	var cntMap [26]int
	for _, char := range s {
		cntMap[char-'a'] += 1
	}
	for idx, char := range s {
		if cntMap[char-'a'] == 1 {
			ans = idx
			break
		}
	}
	return ans
}

func firstUniqChar1(s string) int {
	ans := -1
	cntMap := map[rune]int{}
	for _, char := range s {
		cntMap[char] += 1
	}
	for idx, char := range s {
		if cntMap[char] == 1 {
			ans = idx
			break
		}
	}
	return ans
}

//leetcode submit region end(Prohibit modification and deletion)

File: leetcode/editor/cn/387_字符串中的第一个唯一字符_test.py
import unittest


class TestSolution(unittest.TestCase):

    def setUp(self) -> None:
        self.solution = Solution()

    def test_givenNormal_when_thenSuccess(self):
        self.assertEqual(2, self.solution.firstUniqChar("loveleetcode"))


# leetcode submit region begin(Prohibit modification and deletion)
class Solution:
    def firstUniqChar(self, s: str) -> int:
        ans = -1
        cnt_map = [0] * 26
        for x in s:
            cnt_map[ord(x) - ord('a')] += 1
        for idx, x in enumerate(s):
            if cnt_map[ord(x) - ord('a')] == 1:
                ans = idx
                break
        return ans
    # leetcode submit region end(Prohibit modification and deletion)


if __name__ == '__main__':
    unittest.main()
